static/script/widgets/horizontalcarousel.js

```diff
@@ -19,466 +19,459 @@
+                    throw new Error("HorizontalCarousel::setActiveChildIndex Index out of bounds. " + this.id + " contains " + this._childWidgetOrder.length + " children, but an index of " + index + " was specified.");
+                }
+                return this.setActiveChildWidget(this._childWidgetOrder[index], reposition);
+            },
+            setDataSource: function (data) {
+                this._prefixClones = 0;
+                this._super(data);
+            },
+            rebindDataSource: function () {
+                var device = this.getCurrentApplication().getDevice();
+                var config = device.getConfig();
+                var animate = !config.widgets || !config.widgets.horizontalcarousel || (config.widgets.horizontalcarousel.fade !== false);
+
+                var self = this;
+                var func = this._super;
+                device.hideElement({
+                    el: this._maskElement,
+                    skipAnim: !animate,
+                    onComplete: function () {
+                        func.call(self);
+                    }
+                });
+            },
+            /**
+             * Handle key events to scroll the carousel.
+             * @private
+             */
+            _onKeyDown: function (evt) {
+                // This event handler is already bound (int HorizontalList), we override it to add wrapping logic
+
+                // Block all movement if the carousel is scrolling
+                if (this._scrollHandle && (
+                    evt.keyCode === KeyEvent.VK_LEFT ||
+                        evt.keyCode === KeyEvent.VK_RIGHT ||
+                        evt.keyCode === KeyEvent.VK_UP ||
+                        evt.keyCode === KeyEvent.VK_DOWN
+                    )) {
+                    evt.stopPropagation();
+                    return;
+                }
+
+                switch (evt.keyCode) {
+                    case KeyEvent.VK_LEFT:
+                        if (this.selectPreviousChildWidget()) {
+                            evt.stopPropagation();
+                        }
+
+                        break;
+                    case KeyEvent.VK_RIGHT:
+                        if (this.selectNextChildWidget()) {
+                            evt.stopPropagation();
+                        }
+
+                        break;
+                }
+            },
+
+            /**
+             * DataBound event handler. Clone carousel items to allow infinite scrolling.
+             * @private
+             */
+            _onDataBound: function (/*evt*/) {
+                var application = this.getCurrentApplication();
+                if (!application) {
+                    // application has been destroyed, abort
+                    return;
+                }
+                var device = application.getDevice();
+
+                if (this._childWidgetOrder.length > 0) {
+                    // TODO: tidy up
```

Comment:
I assume this TODO relates to the anonymous function that was underneath it but has now been removed, so it should also be removed.